pkg/terraform/state/deploy.go

```diff
@@ -258,6 +259,9 @@ func (h *handler) createJob(or []metaV1.OwnerReference, input *Input, runName, r
 			Template: coreV1.PodTemplateSpec{
 				Spec: coreV1.PodSpec{
 					ServiceAccountName: sa,
+					ImagePullSecrets: []coreV1.LocalObjectReference{
```

Comment:
If you put ImagePullSecrets into service account used for pod creation, you don't have to provide it in the pod spec:
https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/#add-image-pull-secret-to-service-account

Reply:
I don't see the difference, we create both SA and Job, both are unique per Execution.